### DISCUSSION TITLE
Currently on flex rate, would like to compare agile rate against current usage

### DISCUSSION THREAD

**Discussion by richardambridge** [BottlecapDave/HomeAssistant-OctopusEnergy#376] [Q&A]
I'm currently on flex rate, so fixed all day. This integration works great in HA, but looking to try to add the agile rates into HA so I can run comparisons for the two tariffs. Octopus has a graph you can use, but it only shows "average" rate for the day, so it doesn't account for the fluctuations in the day.

Ideally would like to have a graph in HA that shows,  todays energy cost was X,  if it was on agile it would have been Y.

**Comment by anothermort** [comment]
** not sure if you need to be on agile for this to work **
Is it worth looking at this https://github.com/lozzd/octopus-energy-rates-card
Could this enable you to pull the prices or change it to a graph?

**Comment by BottlecapDave** [comment]
Hi there.

You can use the [tariff override](https://github.com/BottlecapDave/HomeAssistant-OctopusEnergy/blob/develop/_docs/sensors/electricity.md#tariff-overrides) functionality to get the costs of your consumption against a different tariff for the previous day. I'm currently using this as my current tariff is coming to an end and I'm wanting to see which one would be best. 

The charges attribute of these sensors give you the cost for each half hour consumption block, which could be used to plot the rates if these are needed in addition.

You can then use HA's built in statistics graph to plot your actual cost vs the compared cost.

```
chart_type: line
period: hour
type: statistics-graph
entities:
  - >-
    sensor.octopus_energy_electricity_xxx_xxx_previous_accumulative_cost
  - >-
    sensor.octopus_energy_electricity_xxx_xxx_previous_accumulative_cost_override
stat_types:
  - state
```

<img width="368" alt="Screenshot 2023-08-17 170658" src="https://github.com/BottlecapDave/HomeAssistant-OctopusEnergy/assets/14993950/67d0579f-bf1c-4ed2-b32a-f5a2eb64cde8">
